vendor/processors/authorizenet.py

```diff
@@ -324,17 +324,32 @@ def get_transaction_raw_response(self):
             response.pop('messages')
         return str({**self.transaction_message, **response})
 
-    def process_payment_transaction_response(self):
+    def parse_response(self, subscription=True):
         """
-        Processes the transaction reponse from the gateway so it can be saved in the payment model
+        Processes the transaction response from the gateway so it can be saved in the payment model
         """
         self.transaction_id = str(getattr(self.transaction_response, 'transId', 'failed_payment'))
 
-        transaction_info = {}
-        transaction_info['raw'] = self.get_transaction_raw_response()
-        transaction_info['account_type'] = ast.literal_eval(transaction_info['raw']).get('accountType')
+        if not subscription:
+            self.check_response()
+        else:
+            self.check_subscription_response()
+
+        self.transaction_response = self.make_transaction_response(
+            raw=self.get_transaction_raw_response(),
+            messages=f'trans id is {self.transaction_id}'
+
+        )
 
-        self.transaction_response = transaction_info
+    def parse_success(self, subscription=True):
+        if self.transaction_response:
+            if not subscription:
+                if self.transaction_response.messages.resultCode == "Ok":
+                    if hasattr(self.transaction_response.transactionResponse, 'messages') is True:
+                        self.transaction_submitted = True
+            else:
+                if self.transaction_response.messages.resultCode == "Ok":
+                    self.transaction_submitted = True
```

Comment:
I would add this line at the end of the function:
`self.transaction_submitted = False` to be safe